core/iwasm/common/wasm_shared_memory.c

```diff
@@ -380,11 +384,23 @@ static bool
 map_remove_wait_info(HashMap *wait_map_, AtomicWaitInfo *wait_info,
                      void *address)
 {
+    os_mutex_lock(&wait_map_lock);
```

Comment:
`wait_map_` is initialized with true as a second parameter https://github.com/bytecodealliance/wasm-micro-runtime/blob/289fc5efbf5cb2f75f1993ad0131dada78400d3b/core/iwasm/common/wasm_shared_memory.c#L63
which means it already uses a lock internally. Therefore, the lock here should not be necessary.

Currently, `wait_map_lock` is used to make multiple operations on the map atomic (see comments)
https://github.com/bytecodealliance/wasm-micro-runtime/blob/289fc5efbf5cb2f75f1993ad0131dada78400d3b/core/iwasm/common/wasm_shared_memory.c#L125
https://github.com/bytecodealliance/wasm-micro-runtime/blob/289fc5efbf5cb2f75f1993ad0131dada78400d3b/core/iwasm/common/wasm_shared_memory.c#L306

Reply:
Let's maybe add a comment near `wait_map_lock` to specify that the lock is not used for single operations on the map since those are already protected by the internal lock